### PR TITLE
Update linters. Closes #557

### LINKED ISSUE #557: Update linters

### PR DESCRIPTION
**Pull request by tadatuta**
- Update git-hooks and jscs versions
- Update jshint-groups and jscs configs
- Check bemhtml and bemtree with jscs
- Fix all errors

// cc @narqo @dfilatov

### CODE REVIEW

File: common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml [hunk 1]
```diff
@@ -13,7 +15,7 @@ block('i-bem').elem('i18n').def()(function() {
      * Consider `content` is a reserved param that contains
      * valid bemjson data
      */
-    if(ctx.content != null) {
+    if(ctx.content !== null) {
```
Comment (tadatuta) [inline common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml]: @narqo I'm not sure about this change, please check

Comment (dfilatov) [inline common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml]: I think it was made knowingly

Comment (narqo) [inline common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml]: `(ctx.content == null) === (typeof ctx.content === 'undefined' || ctx.content === null)`

Don't really think that `content` could be exact `null` in general, but it helps to filter out other "falsy" values like `0`, `false` etc:

``` js
nul == null  // true
nul == undefined  // true
nul == 0  // false
nul == false  // false
```

File: common.blocks/inherit/inherit.vanilla.js [hunk 2]
```diff
@@ -5,6 +5,8 @@
  * @description This module provides some syntax sugar for "class" declarations, constructors, mixins, "super" calls and static members.
  */
 
+ /* global module, define */
```
Comment (narqo) [inline common.blocks/inherit/inherit.vanilla.js]: WAT? o_O

Comment (tadatuta) [inline common.blocks/inherit/inherit.vanilla.js]: https://github.com/bem/bem-core/blob/issues/557%40v2/common.blocks/inherit/inherit.vanilla.js#L170
https://github.com/bem/bem-core/blob/issues/557%40v2/common.blocks/inherit/inherit.vanilla.js#L181

File: common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml (hunk 1 above)
Comment (tadatuta) [inline common.blocks/i-bem/__i18n/i-bem__i18n.bemhtml]: updated

File: common.blocks/inherit/inherit.vanilla.js (hunk 2 above)
Comment (dfilatov) [inline common.blocks/inherit/inherit.vanilla.js]: I think `inherit` should be excluded from linting as it was done for `vow`.

Comment (tadatuta) [inline common.blocks/inherit/inherit.vanilla.js]: @dfilatov done

**Comment by tadatuta** [comment]
guess everything is ok now?

**Comment by tadatuta** [comment]
@dfilatov, @narqo rebased, let's merge?

**Comment by dfilatov** [comment]
do it )

**Pull request merged**